### DISCUSSION TITLE
Expanding compendium search results

### DISCUSSION THREAD

**Discussion by super-salad** [massif-press/compcon#2131] [Archive]
I've started working on #2077. I've got the Talents and Pilot Gear included in the search results now.

Since there's going to be a lot more data to filter through, I've implemented a 3 character minimum before the search kicks off. Otherwise it loads hundreds of results when you type the first letter only to overwrite them immediately if you keep typing. 

I think 3 characters should be acceptable, although someone wanting to search by "AI" would be unable to. Any ideas if that's a common enough use case to consider? 

I'm just getting all the results to be searchable and rendering now, and will address performance more fully at the end. I'm open to suggestions on speeding it up. A delay in searching until the user finishes typing could be another way to address it as well.

**Comment by msprijatelj** [comment]
"AI" could be common, though it could also be handled via tag. I'd imagine that it would frustrate some users, regardless. I still think the 3 character minimum is fine enough for a start, though.

**Comment by jarena3** [comment]
maybe enforce a three character minimum unless the string is exactly case-insensitive "ai", in which case just search for the AI tag? Weird edge case but that might be the way to go

**Comment by super-salad** [comment]
I am getting spun back up to work on this, and CompCon stuff in general. Sorry for disappearing there for a few months. Life happens.
